components/mpas-ocean/src/mode_forward/mpas_ocn_time_integration_split.F

```diff
@@ -1917,6 +1919,28 @@ subroutine ocn_time_integrator_split(domain, dt)!{{{
       call mpas_pool_begin_iteration(tracersPool)
       do while ( mpas_pool_get_next_member(tracersPool, groupItr) )
          if (groupItr%memberType == MPAS_POOL_FIELD) then
+
+            ! Reset iAge to zero where mask == 0
+            if (trim(groupItr%memberName) == 'idealAgeTracers') then
+                call mpas_pool_get_array(tracersPool, &
+                                         groupItr%memberName, &
+                                         tracersGroupNew, 2)
+                call mpas_pool_get_subpool(forcingPool, &
+                                           'tracersIdealAgeFields', &
+                                            tracersIdealAgeFieldsPool)
+                call mpas_pool_get_array(tracersIdealAgeFieldsPool, &
+                                         'idealAgeTracersIdealAgeMask', &
+                                          tracerGroupIdealAgeMask)
+
+                !$omp parallel
+                !$omp do schedule(runtime)
+                do iCell = 1, nCellsAll
+                   tracersGroupNew(1,1,iCell) = tracerGroupIdealAgeMask(1,iCell)*tracersGroupNew(1,1,iCell)
+                end do ! cells
+                !$omp end do
+                !$omp end parallel
```

Comment:
```suggestion
                if (associated(idealAgeTracersIdealAgeMask)) then
                   !$omp parallel
                   !$omp do schedule(runtime)
                   do iCell = 1, nCellsAll
                      tracersGroupNew(1,1,iCell) = tracerGroupIdealAgeMask(1,iCell)*tracersGroupNew(1,1,iCell)
                   end do ! cells
                   !$omp end do
                   !$omp end parallel
                endif
```
I'm not sure if this is the right way but somehow this needs to not happen when not using ideal age.

Similar changes are needed in `si` module.  It seems like you're not doing this in RK4, so presumably equivalent code needs to be added to the RK4 module.

Reply:
If there's a logical flag for ideal age, I would prefer the checking of a flag rather than checking for association. A lot faster and less reliant on the implementation.

Reply:
@philipwjones, that does sound better.  I'm a bit confused since there are something like 6 different packages associated with `idealAgeTracers` bue presumably there are 1 or more logical flags that could be used instead of the various `associated` checks.